MultiplayerMod/Game/Tools/Events/DebugToolEvents.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using MultiplayerMod.Game.Tools.Context;
+using static DebugTool.Type;
+
+namespace MultiplayerMod.Game.Tools.Events;
+
+public static class DebugToolEvents {
+
+    private static readonly HashSet<DebugTool.Type> ModificationTypes = new() {
+        ReplaceSubstance,
+        FillReplaceSubstance,
+        Clear,
+        Deconstruct,
+        Destroy,
+        StoreSubstance
+    };
+
+    public static event EventHandler<ModifyEventArgs> Modify;
+
+    static DebugToolEvents() => DragToolEvents.DragComplete += OnDragComplete;
+
+    private static void OnDragComplete(object sender, DragCompleteEventArgs e) {
+        if (sender is not DebugTool tool)
+            return;
+
+        if (!ModificationTypes.Contains(tool.type))
+            return;
+
+        var context = new DebugToolContext {
+            AffectCells = DebugPaintElementScreen.Instance.affectCells.isOn,
+            AffectBuildings = DebugPaintElementScreen.Instance.affectBuildings.isOn,
+            PreventFowReveal = DebugPaintElementScreen.Instance.set_prevent_fow_reveal,
+            AllowFowReveal = DebugPaintElementScreen.Instance.set_allow_fow_reveal
+        };
+
+        if (DebugPaintElementScreen.Instance.paintElement.isOn)
```

Comment:
Nit:

Can be moved to constructor with ternary operator.

Also DebugPaintElementScreen.Instance might be extracted to a variable.

Reply:
Nice catch, thanks!